openbb_sdk/providers/fmp/openbb_fmp/models/market_snapshots.py

```diff
@@ -0,0 +1,160 @@
+    "VIE",
+    "WSE",
+    "XETRA",
+]
+
+
+class FMPMarketSnapshotsQueryParams(MarketSnapshotsQueryParams):
+    """FMP Market Snapshots Query.
+
+    Source: https://site.financialmodelingprep.com/developer/docs/#Most-of-the-EuroNext
+    """
+
+    market: MARKETS = Field(
+        description="The market to fetch data for.", default="NASDAQ"
+    )
+
+
+class FMPMarketSnapshotsData(MarketSnapshotsData):
+    """FMP Market Snapshots Data."""
+
+    class Config:
+        fields = {
+            "high": "dayHigh",
+            "low": "dayLow",
+            "prev_close": "previousClose",
+            "change_percent": "changesPercentage",
+        }
+
+    avg_volume: Optional[int] = Field(
+        description="Average volume of the stock.", alias="avgVolume"
+    )
+    ma50: Optional[float] = Field(
+        description="The 50-day moving average.", alias="priceAvg50"
+    )
+    ma200: Optional[float] = Field(
+        description="The 200-day moving average.", alias="priceAvg200"
+    )
+    year_high: Optional[float] = Field(
+        description="The 52-week high.", alias="yearHigh"
+    )
+    year_low: Optional[float] = Field(description="The 52-week low.", alias="yearLow")
+    market_cap: Optional[float] = Field(
+        description="Market cap of the stock.", alias="marketCap"
+    )
+    shares_outstanding: Optional[int] = Field(
+        description="Number of shares outstanding.", alias="sharesOutstanding"
+    )
+    eps: Optional[float] = Field(description="Earnings per share.")
+    pe: Optional[float] = Field(description="Price to earnings ratio.")
+    exchange: Optional[str] = Field(description="The exchange of the stock.")
+
+
+class FMPMarketSnapshotsFetcher(
+    Fetcher[
+        FMPMarketSnapshotsQueryParams,
+        List[FMPMarketSnapshotsData],
+    ]
+):
+    """Transform the query, extract and transform the data from the FMP endpoints."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> FMPMarketSnapshotsQueryParams:
+        """Transform the query params."""
+        return FMPMarketSnapshotsQueryParams(**params)
+
+    @staticmethod
+    def extract_data(
+        query: FMPMarketSnapshotsQueryParams,
+        credentials: Optional[Dict[str, str]],
+        **kwargs: Any,
+    ) -> List[Dict]:
+        """Return the raw data from the FMP endpoint."""
+
+        api_key = credentials.get("fmp_api_key") if credentials else ""
+        url = f"https://financialmodelingprep.com/api/v3/quotes/{query.market}?apikey={api_key}"
+        response = make_request(url)
+
+        if response.status_code != 200:
+            raise RuntimeError(
+                f"Error fetching data from FMP  -> {response.status_code}"
+            )
+        data = pd.DataFrame(response.json())
+        data = data.drop(columns=["timestamp", "earningsAnnouncement"]).fillna(value=0)
+        data = data[data["price"] > 0.01]
+
+        return data.sort_values(by="volume", ascending=False).to_dict(orient="records")
```

Comment:
The line above that, however, removes a lot of terrible noise in the data set by filtering for prices above 1 cent.

Reply:
@IgorWounds thoughts?

Reply:
In line with my other comment, we don't filter the provider's response and just return the standard + extra fields. The provider is the one responsible for the "terrible noise" and it should be transparent so that our users can easily cherry-pick the right providers. Some might be even looking for that noise and similar as it indicates market/provider inefficiency that could be used in one way or another.

Reply:
Revised.